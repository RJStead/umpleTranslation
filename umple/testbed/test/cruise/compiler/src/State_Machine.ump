generate Java "../src-gen-umple";

class LightFixture  
{  
  bulb  
  {  
    On {  
      entry / { doEntry(); }  
      exit / { doExit(); }  
      push -> /{ doTransition(); } Off;  
      do { doThisContinuouslyWhileOn(); a++; }  
    }  
    Off {}  
  }  
  
  void doEntry() {a=1;}  
  void doExit() {System.out.println("Exit");}  
  void doTransition() {System.out.println("Transition");}  
  void doThisContinuouslyWhileOn() {  
    while (true) {  
      System.out.println("Still on");  
      try {  
        Thread.sleep(1000);  
      }  
      catch (InterruptedException e) {}  
    }  
  }  
}
