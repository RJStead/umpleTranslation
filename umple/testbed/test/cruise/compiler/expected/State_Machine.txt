State_Machine.ump:11: error: cannot find symbol
      doThisContinuouslyWhileOn(); a++;
                                   ^
  symbol:   variable a
  location: class LightFixture
State_Machine.ump:16: error: cannot find symbol
    a=1;
    ^
  symbol:   variable a
  location: class LightFixture
2 errors